exchanges/deribit/deribit_test.go

```diff
@@ -0,0 +1,2397 @@
+		Trigger: "", Advanced: "", Amount: info.ContractSize * 3,
+		Price: 500000, MaxShow: 0, TriggerPrice: 0, PostOnly: false,
+		RejectPostOnly: false, ReduceOnly: false, MMP: false})
+	assert.NoError(t, err)
+}
+
+func TestEditOrderByLabel(t *testing.T) {
+	t.Parallel()
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, d, canManipulateRealOrders)
+	_, err := d.EditOrderByLabel(context.Background(), &OrderBuyAndSellParams{Label: "incorrectUserLabel", Instrument: btcPerpInstrument,
+		Advanced: "", Amount: 1, Price: 30000, TriggerPrice: 0, PostOnly: false, ReduceOnly: false, RejectPostOnly: false, MMP: false})
+	assert.NoError(t, err)
+	_, err = d.WSEditOrderByLabel(&OrderBuyAndSellParams{Label: "incorrectUserLabel", Instrument: btcPerpInstrument,
+		Advanced: "", Amount: 1, Price: 30000, TriggerPrice: 0, PostOnly: false, ReduceOnly: false, RejectPostOnly: false, MMP: false})
+	assert.NoError(t, err)
+}
+
+const submitCancelResponseJSON = `{"triggered": false, "trigger": "index_price", "time_in_force": "good_til_cancelled", "trigger_price": 144.73, "reduce_only": false, "profit_loss": 0, "price": "market_price", "post_only": false, "order_type": "stop_market", "order_state": "untriggered", "order_id": "ETH-SLIS-12", "max_show": 5, "last_update_timestamp": 1550575961291, "label": "", "is_liquidation": false, "instrument_name": "ETH-PERPETUAL", "direction": "sell", "creation_timestamp": 1550575961291, "api": false, "amount": 5 }`
+
+func TestSubmitCancel(t *testing.T) {
+	t.Parallel()
+	var resp *PrivateCancelData
+	err := json.Unmarshal([]byte(submitCancelResponseJSON), &resp)
+	if err != nil {
+		t.Fatal(err)
+	}
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, d, canManipulateRealOrders)
+	_, err = d.SubmitCancel(context.Background(), "incorrectID")
+	assert.NoError(t, err)
+	_, err = d.WSSubmitCancel("incorrectID")
+	assert.NoError(t, err)
+}
+
+func TestSubmitCancelAll(t *testing.T) {
+	t.Parallel()
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, d, canManipulateRealOrders)
+	_, err := d.SubmitCancelAll(context.Background())
+	assert.NoError(t, err)
+	_, err = d.WSSubmitCancelAll()
+	assert.NoError(t, err)
+}
+
+func TestSubmitCancelAllByCurrency(t *testing.T) {
+	t.Parallel()
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, d, canManipulateRealOrders)
+	_, err := d.SubmitCancelAllByCurrency(context.Background(), currency.BTC, "option", "")
+	assert.NoError(t, err)
+	_, err = d.WSSubmitCancelAllByCurrency(currencyBTC, "option", "")
+	assert.NoError(t, err)
+}
+
+func TestSubmitCancelAllByKind(t *testing.T) {
+	t.Parallel()
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, d, canManipulateRealOrders)
+	_, err := d.SubmitCancelAllByKind(context.Background(), currency.ETH, "option_combo", "trigger_all", true)
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = d.WsSubmitCancelAllByKind(currency.ETH, "option_combo", "trigger_all", true)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestSubmitCancelAllByInstrument(t *testing.T) {
```

Comment:
Error:      	Received unexpected error:
        	            	json: cannot unmarshal array into Go value of type deribit.CancelResponse
        	Test:       	TestSubmitCancelAllByInstrument

This one might just be optional since I have no funds

Reply:
Me too :) ... I couldn't verify it because I am not getting the correct result.